frontend/src/general/__tests__/DynamicBreadcrumb.test.jsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import Enzyme from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+import { afterEach, beforeEach, describe, expect, it } from "@jest/globals";
+
+import { Box, Breadcrumbs, Link, Typography } from "@material-ui/core";
+import { NavLink } from "react-router-dom";
+import { NavigateNext } from "@material-ui/icons";
+import DeviceInfo from "../../model/DeviceInfo";
+import StreamInfo from "../../model/StreamInfo";
+import DynamicBreadcrumb from "../DynamicBreadcrumb";
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe("<DynamicBreadcrumb/> functional component", () => {
+  let wrapper;
+  const objectBreadcrumb = [
+    "someDisplayName",
+    "somePathName",
+    { beep: new DeviceInfo("beep"), boop: new StreamInfo("boop") }
+  ];
+  const noObjectBreadcrumb = ["someOtherDisplayName", "someOtherPathName"];
+  const breadcrumbs = [objectBreadcrumb, noObjectBreadcrumb];
+
+  beforeEach(() => {
+    wrapper = Enzyme.shallow(<DynamicBreadcrumb breadcrumbs={breadcrumbs} />);
+  });
+  afterEach(() => {
+    wrapper.unmount();
+  });
+  describe("returns a component that", () => {
+    it("Contains one <Box/> component with expected props", () => {
+      const components = wrapper.find(Box);
+      expect(components).toHaveLength(1);
+
+      const props = components.at(0).props();
+
+      const expected = {
+        padding: "2em 0em 0em"
+      };
+      expect(props.padding).toBe(expected.padding);
+    });
+    it("Contains one <Breadcrumbs/> component with expected props", () => {
+      const components = wrapper.find(Breadcrumbs);
+      expect(components).toHaveLength(1);
+
+      const props = components.at(0).props();
+
+      const expected = {
+        "aria-label": "breadcrumb",
+        separator: <NavigateNext fontSize="medium" />
+      };
+      expect(props).toMatchObject(expected);
+    });
+    it("Contains <Link/> components equal to number of crumbs passed", () => {
+      const components = wrapper.find(Link);
+      expect(components).toHaveLength(breadcrumbs.length);
+    });
+    it("<Link/> 0 has expected props", () => {
+      const components = wrapper.find(Link);
+
+      const props = components.at(0).props();
+
+      const expected = {
+        component: NavLink,
+        to: {
+          pathname: objectBreadcrumb[1],
+          state: objectBreadcrumb[2]
```

Comment:
is this getting 
```
state : {
  beep: (deviceInfo object),
  boop: (streamInfo object)
}
```
? just to make sure i understand xD

Reply:
yeah. It was designed like this because depending on the page, the state will require different format 
eg: Device details looks for `location.state.device` whereas stream details looks for `location.state.stream` 

and potentially others such as create stream w/ this channel to send the channel & view my senders/receivers to send  if it's the senders tab or receivers tab